src/clj/puppetlabs/services/jruby_pool_manager/impl/jruby_internal.clj

```diff
@@ -86,6 +86,29 @@
    pool :- jruby-schemas/pool-queue-type]
   (.borrowItemWithTimeout pool timeout TimeUnit/MILLISECONDS))
 
+(schema/defn insert-shutdown-poison-pill
+  [pool :- jruby-schemas/pool-queue-type]
+  (.insertPill pool (ShutdownPoisonPill. pool)))
+
+(schema/defn insert-poison-pill
+  [error :- Throwable
+   pool :- jruby-schemas/pool-queue-type]
+  (.insertPill pool (PoisonPill. error)))
+
+(schema/defn fill-pool-with-poison-pills!
+  "Fill an empty pool with poison pills. Should always be run on the
+  modify-instance-agent. insert-fn should be a fn that inserts a poison
+  pill of some kind into the pool"
+  [pool-state :- jruby-schemas/PoolState
+   insert-fn :- IFn]
+  (let [pool (:pool pool-state)
+        pool-size (:size pool-state)]
+    (when (not (= pool-size (.remainingCapacity pool)))
```

Comment:
Oh, in that case, I think it's 💯  to throw an Exception since the lock will never be available anyway.